File: RxStore.ForRazor/EventTransition.cs
namespace RxStore
{
    public record EventTransition<TVersion, TState, TEvent>(
        TVersion Version,
        TState State,
        TEvent Event
    );


    public static class EventTransition
    {
        public static EventTransition<TVersion, TState, TEvent> Of<TVersion, TState, TEvent>(
            TVersion version,
            TState state,
            TEvent @event
        )
        {
            return new EventTransition<TVersion, TState, TEvent>(version, state, @event);
        }
    }
}

File: RxStore.Entity/EventTransition.fs
namespace RxStore


type EventTransition<'State, 'Event> =
    { State: 'State;
      Event: 'Event }


module EventTransition =

    [<CompiledName("OfValues")>]
    let ofValues state event =
        { State = state;
          Event = event }

    [<CompiledName("AsUpdateState")>]
    let ofTransitionState<'State, 'Event>
        (transition: EventTransition<'State, 'Event>)
        : EventUpdate<'State, 'Event> =
            State transition.State

    [<CompiledName("AsUpdateEvent")>]
    let ofTransitionEvent<'State, 'Event>
        (transition: EventTransition<'State, 'Event>)
        : EventUpdate<'State, 'Event> =
            Event transition.Event

    [<CompiledName("AsUpdateConfirmation")>]
    let ofTransitionConfirmation<'State, 'Event>
        (transition: EventTransition<'State, 'Event>)
        : EventUpdate<'State, 'Event> =
            Confirmation
